### DISCUSSION TITLE
Custom fetch replacement?

### DISCUSSION THREAD

**Discussion by sebastianlp** [connectrpc/connect-es#489] [Q&A] [answered]
Hi guys! 
First of all a big congrats for this great job, I've been following the blog posts since the beginning and some days ago I have the chance to experiment with connect-web, really happy so far 😄 
I have a question which may have already been answered, I have a custom axios implementation which handles authentication and retries policies that I want to use instead of `fetch`. 
I was thinking to create my own `connect-transport` that uses that axios instance but want to know your opinion on this or if you have a better solution?

I've seen a PR and an issue that discuss passing a custom fetch instance but nothing related to axios for now... maybe that mentioned issue/pr could be enhanced to allow even axios instances or other http clients?

**Comment by timostamm** [comment, marked as answer]
Hey Sebastián, writing your own [Transport](https://github.com/bufbuild/connect-es/blob/199e29e27f8e0d2ab66c3f6c83d50b863cff4de5/packages/connect/src/transport.ts#L29) is straight-forward. The [implementation with fetch](https://github.com/bufbuild/connect-es/blob/199e29e27f8e0d2ab66c3f6c83d50b863cff4de5/packages/connect-web/src/connect-transport.ts#L113) is just 200 lines of code. Only 80 lines if you don't care about streaming. So creating your own Connect transport that uses axios is definitely a feasible option!

Are you aware of [axios-fetch](https://github.com/lifeomic/axios-fetch)? It turns an axios instance into a fetch function. Have not tried it, but this could be another approach once we have a solution for https://github.com/bufbuild/connect-es/issues/403.

**Comment by sebastianlp** [comment]
Hey Timo thanks for your answer. I wasn't aware of [axios-fetch](https://github.com/lifeomic/axios-fetch) but unfortunately I can't use it because I don't have access to the axios instance, I just have a wrapper on top of it from a dependency 😞 
Will probably take writing the own transport option for now
